RxSwift/Traits/Single.swift

```diff
@@ -0,0 +1,226 @@
+     */
+    public static func just(_ element: ElementType, scheduler: ImmediateSchedulerType) -> Single<ElementType> {
+        return Single(raw: Observable.just(element, scheduler: scheduler))
+    }
+    
+    /**
+     Returns an observable sequence by the source observable sequence shifted forward in time by a specified delay. Error events from the source observable sequence are not delayed.
+     
+     - seealso: [delay operator on reactivex.io](http://reactivex.io/documentation/operators/delay.html)
+     
+     - parameter dueTime: Relative time shift of the source by.
+     - parameter scheduler: Scheduler to run the subscription delay timer on.
+     - returns: the source Observable shifted in time by the specified delay.
+     */
+    public func delay(_ dueTime: RxTimeInterval, scheduler: SchedulerType)
+        -> Single<ElementType> {
+            return Single(raw: primitiveSequence.source.delay(dueTime, scheduler: scheduler))
+    }
+    
+    /**
+     Invokes an action for each event in the observable sequence, and propagates all observer messages through the result sequence.
+     
+     - seealso: [do operator on reactivex.io](http://reactivex.io/documentation/operators/do.html)
+     
+     - parameter onNext: Action to invoke for each element in the observable sequence.
+     - parameter onError: Action to invoke upon errored termination of the observable sequence.
+     - parameter onSubscribe: Action to invoke before subscribing to source observable sequence.
+     - parameter onSubscribed: Action to invoke after subscribing to source observable sequence.
+     - parameter onDispose: Action to invoke after subscription to source observable has been disposed for any reason. It can be either because sequence terminates for some reason or observer subscription being disposed.
+     - returns: The source sequence with the side-effecting behavior applied.
+     */
+    public func `do`(onNext: ((ElementType) throws -> Void)? = nil,
+                     onError: ((Swift.Error) throws -> Void)? = nil,
+                     onSubscribe: (() -> ())? = nil,
+                     onSubscribed: (() -> ())? = nil,
+                     onDispose: (() -> ())? = nil)
+        -> Single<ElementType> {
+            return Single(raw: primitiveSequence.source.do(
+                onNext: onNext,
+                onError: onError,
+                onSubscribe: onSubscribe,
+                onSubscribed: onSubscribed,
+                onDispose: onDispose)
+            )
+    }
+    
+    /**
+     Filters the elements of an observable sequence based on a predicate.
+     
+     - seealso: [filter operator on reactivex.io](http://reactivex.io/documentation/operators/filter.html)
+     
+     - parameter predicate: A function to test each source element for a condition.
+     - returns: An observable sequence that contains elements from the input sequence that satisfy the condition.
```

Comment:
I think we can leave `observable sequence` terminology. It will also make our lives easier while we will try to keep documentation consistent, and `Maybe` is an observable sequence with additional guarantees IMHO.